### PR TITLE
Add metrics to all locks/semaphores

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
In order to get insights on time spent on locks in rnode add metrics to all locks/semaphores.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3766

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala [hunk 1]
```diff
@@ -44,25 +44,14 @@ package object accounting extends Costs {
...
-                    )
-              } yield ()
-          ) { _ =>
-            cost.release
-          }
-      _ <- error.ensure(cost.inspect(identity))(OutOfPhlogistonsError)(_.value >= 0)
-    } yield ()
+    cost.withPermit(
+      cost.get.flatMap { c =>
+        Monad[F].ifM[Unit]((c.value < 0).pure[F])(
```
Comment (ArturGajowy) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala]: :ok_hand: Maybe just an `if` instead of `ifM` though?

**Review by ArturGajowy — APPROVED** [review]
Rholang part LGTM

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala [hunk 2]
```diff
@@ -44,33 +44,20 @@ package object accounting extends Costs {
...
-                        _        <- cost.set(newValue)
-                      } yield (())
-                    )
-              } yield ()
-          ) { _ =>
-            cost.release
-          }
-      _ <- error.ensure(cost.inspect(identity))(OutOfPhlogistonsError)(_.value >= 0)
-    } yield ()
+    cost.withPermit(
```
Comment (lukasz-golebiewski) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala:47]: Nice job, looks much cleaner!

**Review by lukasz-golebiewski — APPROVED** [review]

**Comment by dzajkowski** [comment]
bors try

**Review by itegulov — APPROVED** [review]
Blockstorage LGTM

**Pull request merged**